src/api/License/index.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Linkurious SAS 2012 - 2022
+ *
+ * - Created on 2022-03-16.
+ */
+import {Request} from '../../http/request';
+import {LkErrorKey} from '../../http/response';
+
+import {ISubmitLicenseParams, LicenseState} from './types';
+
+const {INVALID_LICENSE} = LkErrorKey;
+
+export class LicenseAPI extends Request {
+  /**
+   * Verify the supplied license and save it if it’s valid and there is no saved license yet.
+   */
+  public submitLicense(this: Request<LicenseState>, params: ISubmitLicenseParams) {
```

Comment:
When this API will be called? I thought that the license will be saved when we call setupAuthentication and not the first call

Reply:
This API has to be called if the license is missing (check the `license` value of the `GET /api/config` response). There are cases where the setupAuthentication endpoint won't be called (for instance if authentication is disabled); that's why this endpoint saves the uploaded license if there is no installed license yet.

If setupAuthentication is called without calling this endpoint first, it will also save the license with the same condition (internally, both endpoints call the same license service method). Submitting the license twice is not a big deal as it's a very small file, and the license checking/saving logic is idempotent.